lib/prelude.dx

```diff
@@ -2469,3 +2535,15 @@ instance {a b} [Ix a, Ix b] Ix (a=>b)
   size = intpow (size b) (size a)
   ordinal             = reversed_digits_to_int
   unsafe_from_ordinal = int_to_reversed_digits
+
+instance {a b} [NonEmpty a, NonEmpty b] NonEmpty (a=>b)
+  first_ix = unsafe_from_ordinal _ 0
+
+instance {a b} [NonEmpty a, Ix b, Subset a b] NonEmpty b
+  first_ix = unsafe_from_ordinal _ 0
```

Comment:
I'm a bit afraid this instance will cause some "duplicate instance" errors 😕

Reply:
Good point, will remove.